### PR TITLE
Enforce limits of values read from incoming headers and app id lookup

### PR DESCRIPTION
**Pull request by cijothomas**
Addresses security concerns about malicious user attempting to send request with unreasonably large request headers. As SDK reads these values and stores locally/make part of Telemetry items, they can cause undesirable effects like high mem/cpu/ etc.
This attempts to enforce limits on values read from outside requests/responses.

- [ ] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Changes in public surface reviewed
- [ ] Design discussion issue #
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue.
- [ ] The PR will trigger build, unit tests, and functional tests automatically. If your PR was submitted from fork - mention one of committers to initiate the build for you.
	  If you want to to re-run the build/tests, the easiest way is to simply Close and Re-Open this same PR. (Just click 'close pull request' followed by 'open pull request' buttons at the bottom of the PR)

- Please follow [these] (https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/develop/Readme.md) instructions to build and test locally.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNetCore/Common/InjectionGuardConstants.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        /// <summary>
+        /// Max length of incoming Request Header value allowed.
+        /// </summary>
+        public const int RequestHeaderMaxLength = 100;
+
+        /// <summary>
+        /// Max length of context header key.
+        /// </summary>
+        public const int ContextHeaderKeyMaxLength = 50;
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights.AspNetCore/Common/InjectionGuardConstants.cs]: i picked 50. Not sure if there is genuine need of key longer than this.

File: src/Microsoft.ApplicationInsights.AspNetCore/Common/InjectionGuardConstants.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        /// <summary>
+        /// Max length of context header value.
+        /// </summary>
+        public const int ContextHeaderValueMaxLength = 100;
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights.AspNetCore/Common/InjectionGuardConstants.cs]: i picked 100. Not sure if there is genuine need of a value longer than this.

Comment (MS-TimothyMothra) [reply src/Microsoft.ApplicationInsights.AspNetCore/Common/InjectionGuardConstants.cs]: @SergeyKanzhelev's guidance for WebSDK was to allow up to 1Kb. I made the same change above to `RequestHeaderMaxLengeth`.
Consider if that is appropriate for here as well.

**Comment by MS-TimothyMothra** [comment]
I'm approving these changes. Most everything was copied verbatim from https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/810

**Pull request merged**